Pyrado/pyrado/environments/mujoco/openai_ant.py

```diff
@@ -0,0 +1,175 @@
+        super().__init__(model_path, frame_skip, dt, max_steps, task_args)
+
+        # Initial state
+        noise_halfspan = self.domain_param["reset_noise_halfspan"]
+        min_init_qpos = self.init_qpos - np.full_like(self.init_qpos, noise_halfspan)
+        max_init_qpos = self.init_qpos + np.full_like(self.init_qpos, noise_halfspan)
+        min_init_qvel = self.init_qvel - np.full_like(self.init_qvel, noise_halfspan)
+        max_init_qvel = self.init_qvel + np.full_like(self.init_qvel, noise_halfspan)
+
+        cfrc_shape = self.sim.data.cfrc_ext.flat.copy().shape
+        min_init_cfrc = -np.ones(cfrc_shape)
+        max_init_cfrc = np.ones(cfrc_shape)
+        min_init_state = np.concatenate([min_init_qpos, min_init_qvel, min_init_cfrc]).ravel()
+        max_init_state = np.concatenate([max_init_qpos, max_init_qvel, max_init_cfrc]).ravel()
+        self._init_space = BoxSpace(min_init_state, max_init_state)
+
+        self.camera_config = dict(distance=5.0)
+
+    @property
+    def state_space(self) -> Space:
+        state_shape = np.concatenate(
+            [self.sim.data.qpos.flat, self.sim.data.qvel.flat, self.sim.data.cfrc_ext.flat]
+        ).shape
+        return BoxSpace(-pyrado.inf, pyrado.inf, shape=state_shape)
+
+    @property
+    def obs_space(self) -> Space:
+        obs_shape = self.observe(self.state_space.bound_up).shape
+        return BoxSpace(-pyrado.inf, pyrado.inf, shape=obs_shape)
+
+    @property
+    def act_space(self) -> Space:
+        act_bounds = self.model.actuator_ctrlrange.copy().T
+        return BoxSpace(
+            *act_bounds, labels=["hip_4", "ankle_4", "hip_1", "ankle_1", "hip_2", "ankle_2", "hip_3", "ankle_3"]
+        )
+
+    @classmethod
+    def get_nominal_domain_param(cls) -> dict:
+        return dict(
+            reset_noise_halfspan=0.1,
+            init_pos_z=0.75,
+            hip_length=0.2,
+            thigh_length=0.2,
+            tibia_length=0.4,
+            gravity=9.81,
+            sliding_friction=1,
+            torsional_friction=0.5,
+            rolling_friction=0.5,
+            density=5.0,  # scales linearly with the mass
+            wind_x=0,
+            wind_y=0,
+            wind_z=0,
+        )
+
+    def _create_task(self, task_args: dict) -> Task:
+        # Define the task including the reward function
+        if "contact_cost_weight" not in task_args:
+            task_args["contact_cost_weight"] = 5e-4
+        if "ctrl_cost_weight" not in task_args:
+            task_args["ctrl_cost_weight"] = 0.5
+        if "healthy_reward" not in task_args:
+            task_args["healthy_reward"] = 1.0
+        if "terminate_when_unhealthy" not in task_args:
+            task_args["terminate_when_unhealthy"] = True
+        if "healthy_z_range" not in task_args:
+            task_args["healthy_z_range"] = (0.2, 1.0)
+        if "contact_force_range" not in task_args:
+            task_args["contact_force_range"] = (-1.0, 1.0)
```

Comment:
I would suggest to move these default values to the constructor of the reward function